### ISSUE TITLE
Training Pipeline and Loss function

### ISSUE THREAD

**Issue by YHRen** [LS4GAN/ls4gan-python#3]
### Description

- [ ] A cGAN training pipeline taking models, data and hyper-parameters such as learning rate and batch size as `ls4gan/train/trainer.py`.
- [ ] A Loss function collection as `ls4gan/train/losses.py`
